### DISCUSSION TITLE
How good is DomPDF?

### DISCUSSION THREAD

**Discussion by prodigy7** [dompdf/dompdf#2944] [General]
Hello all!

I've been programming in PHP for a long time and accordingly had to deal with the generation of PDF files often enough.
I started with fpdf, then I ended up with snappy which is a wrapper for wkhtmltopdf and now I came across your project.

What I'm wondering now: How good is DomPDF?

fpdf was a disaster when it came to creating pdf files.
wkhtmltopdf was fantastic in many respects, but is probably not really being developed any further.

Now I'm looking around to see where the journey can take me and I've landed on your site. Can I achieve at least as good results with your library as with wkhtmltopdf? What is better, what is worse? I'd be interested to hear a comparison from those of you who have experience.

p7

**Comment by bsweeney** [comment]
I'd say generally Dompdf provides a solid experience for generating a PDF from an HTML + CSS document. While Dompdf can't compete with a native binary like wkhtmltopdf it does have pretty good [CSS support](https://github.com/dompdf/dompdf/wiki/CSSCompatibility) and is decently performant with recent releases on recent versions of PHP.

Whether or not it's worth pursuing entirely depends on the needs for your specific project. There are some quirks ([rendering order](2870), [tables](https://github.com/dompdf/dompdf/issues/98)) and missing features ([flexbox](https://github.com/dompdf/dompdf/issues/971), [box-sizing](https://github.com/dompdf/dompdf/issues/669), [internationalization](https://github.com/dompdf/dompdf/issues/2619)). It's worth looking through outstanding issues to see if there's anything that would be a blocker for your projects.

As far as maintenance and development, there's no permanent team in place. We're in a good place now as far as moving the project forward, but Dompdf is not a funded project and as such future development is entirely dependent on the good will of the persons involved. This means a schedule for bug fixes and feature development can't be provided.
